src/Microsoft.ML.Core/Utilities/Hashing.cs

```diff
@@ -68,17 +68,19 @@ public static int HashInt(int n)
         public static uint HashString(string str)
         {
             Contracts.AssertValue(str);
-            return MurmurHash((5381 << 16) + 5381, str, 0, str.Length);
+            return MurmurHash((5381 << 16) + 5381, str.AsMemory());
         }
 
+        public static uint HashString(ReadOnlyMemory<char> str) => MurmurHash((5381 << 16) + 5381, str);
+
         /// <summary>
         /// Hash the characters in a sub-string. This MUST produce the same result
         /// as HashString(str.SubString(ichMin, ichLim - ichMin)).
         /// </summary>
         public static uint HashString(string str, int ichMin, int ichLim)
         {
             Contracts.Assert(0 <= ichMin & ichMin <= ichLim & ichLim <= Utils.Size(str));
-            return MurmurHash((5381 << 16) + 5381, str, ichMin, ichLim);
+            return MurmurHash((5381 << 16) + 5381, str.AsMemory().Slice(ichMin, ichLim - ichMin));
```

Comment:
>str.AsMemory() [](start = 51, length = 14)

`.AsMemory` is a pretty horrifyingly slow operation from my timings of it. I hope we aren't calling this too often.

Reply:
We should take this issue up with the .Net team. I don't understand why ReadOnlyMemory<char> cannot be instantiated with a string? It does have an implicit operator for assignment with T[].

---
In reply to: [216127951](https://github.com/dotnet/machinelearning/pull/863#discussion_r216127951) [](ancestors = 216127951)